lib/logstash/util.rb

```diff
@@ -24,6 +24,37 @@ def self.set_thread_name(name)
     end
   end # def set_thread_name
 
+  def self.set_thread_plugin(plugin)
+    Thread.current[:plugin] = plugin
+  end
+
+  def self.get_thread_id(thread)
+    if RUBY_ENGINE == "jruby"
+      JRuby.reference(thread).native_thread.id
+    else
+      raise Exception.new("Native thread IDs aren't supported outside of JRuby")
+    end
+  end
+
+  def self.thread_info(thread)
+    backtrace = thread.backtrace.map do |line|
+      line.gsub(LogStash::Environment::LOGSTASH_HOME, "[...]")
+    end
+    state = case backtrace.first
+            when /in `push'/ then "blocked_on_push"
+            when /(?:pipeline|base).*pop/ then "waiting_for_events"
+            else "running"
+            end
```

Comment:
We need test harness for the inspected string, since this part of the code will change in the upcoming weeks.